### DISCUSSION TITLE
Unable to see Pool as Main Body #344

### DISCUSSION THREAD

**Discussion by vj950** [tagyoureit/nodejs-poolController#436] [App and O/S Config Questions] [answered]
Previously resolved issue #344 is back when I re-opened the pool after a hiatus of 6 months and had to re-install everything on pi zero after SD card wipe. I think it might have happened with code changes with new updates. 

Per Gitter instruction, I am attaching replay of packet capture and timestamp. Also attaching the screenshot of the dashpanel which shows that the "Pool" circuit is missing. The first schedule (red arrow) should say "Pool" from my screenshot from previous fix, but is nameless. The Pentair setup is exactly the same as previous
![Dashpanel](https://user-images.githubusercontent.com/62803887/167326298-c6011e15-bb10-4cbd-8cea-09b7ec971dbf.JPG)
![main3](https://user-images.githubusercontent.com/62803887/167326320-e577f6e9-c124-4f9e-bbf3-4bad3d91664d.png)
![timestamps3](https://user-images.githubusercontent.com/62803887/167326352-34fe1ea7-da25-4844-a47d-e41d924672e8.JPG)
[replay(3).zip](https://github.com/tagyoureit/nodejs-poolController/files/8648247/replay.3.zip)
.

**Comment by rstrouse** [comment]
Pull njsPC.  I think we have been getting reports that are incorrect other IntelliTouch iXS users.  This system is really no different than a shared body system but it has the intake and return valves disabled.  So there are two bodies on a single body system for this type of controller.

**Comment by vj950** [comment]
The changes added the pool circuit (id1). However, the njpc still can't assign the heater properly. Therefore, the heater is assigned to the Solar (id6). Please see the screenshot
![Capture4](https://user-images.githubusercontent.com/62803887/167753839-103b9f8b-7f3e-4f10-8f00-e2649b8be1fc.JPG)

BTW, thank you very much from mqtt fix, I now can control all the circuits from Home Assistant.

**Comment by rstrouse** [comment]
Go to the heater config and assign it to both bodies.

**Comment by vj950** [comment]
I am not able to assign in heater menu. Capture 1 and 2 shows heater options. In capture 2 "Pool" is actually Solar. Hi-Temp showed up as option after the code fix couple of days ago. If I select it and save, the "saving" dialogue stays there forever, nothing gets saved.
![Heater1](https://user-images.githubusercontent.com/62803887/167971083-58d00a4f-2615-4b2b-b1c0-c5ba977a08e1.JPG)


![Heater3](https://user-images.githubusercontent.com/62803887/167971119-1c75ff26-a9f0-449c-acdf-deadbd5077f7.JPG)

"Hi-Temp" is correctly associated with Pool. Low temp is Solar. The Heater for Pool is On in dashpanel, but I was only able to turn it on using Remote. As we see from panel, the setpoint of 71 degrees is displayed (solar circuit), the pool heater (or Hi-Temp) setpoint of 75 degrees is not seen.
![20220511_210743](https://user-images.githubusercontent.com/62803887/167972828-5212039b-87ca-40b8-809f-44cdaf4b832c.jpg)
![panel](https://user-images.githubusercontent.com/62803887/167972849-7dfb7162-1f59-4d1f-9ce7-4b50c48b4460.JPG)

I think Heater is not recognized by njpc is because in Pump menu, Heater should be shown where the red arrow is, because that's what shows up on OCP remote.
![Pump](https://user-images.githubusercontent.com/62803887/167971244-8761779c-04e0-479d-8d78-db65faf09485.JPG)

I think we still have trouble with circuits, because on remote, Solar is set as Master Pool and Pool is set as Master Spa. It shows up as in screenshot. 
![20211006_210238](https://user-images.githubusercontent.com/62803887/167971358-c11431fe-76dd-49ec-84ae-489e2aa3fb32.jpg)
![20211006_210314](https://user-images.githubusercontent.com/62803887/167971377-463a84c9-5640-45d8-8b1f-93949b4d0673.jpg)

We nailed everything correctly back in October

**Reply by rstrouse** [reply]
Can I get another replay now that you updated your system.  The update should have enabled the heater on both bodies.  I have fixed the virtual circuits (the heater not showing in the dropdown) but still need to make sure we have the right byte mapping on the status message.

**Comment by rstrouse** [comment]
Go ahead and pull njsPC.  I found a couple of places where we may have gotten some bad data from your OCP.  njsPC will now detect these conditions and clean them up instead of choking them down.

**Comment by vj950** [comment]
pulled the latest njsPC. The "Pool" is now recognized as "Spa", which is more accurate.
![2022-05-18-1](https://user-images.githubusercontent.com/62803887/169185124-b0215c32-8565-46fa-adae-3b986aa96146.JPG)


 Still unable to assign "Spa" in Gas Heater section, as the changes are not saved.
![2022-05-18-2](https://user-images.githubusercontent.com/62803887/169185299-7f9daf6d-6a66-4a31-883c-ac8bd9de0c1d.JPG)

This time, I experimented with types of heaters available under "Assign custom speeds" in Pump setup menu on Remote, and set "Pool Heater" at 2100 RPM, and "Spa Heater" at 2050 RPM. The originally set "Heater" in this menu was still present, set at 2000 RPM. Interestigly, the pump recognized the type of heater activated this time. On all previous occasions, I only had one Heater as Heater with RPM of 2000. The left capture is  with 3 separate heater speeds set on pump. On the right is when I removed those custom speeds from Pump menu on remote and went with my original configuration of only one Heater.

Needless to say, on njsPC dashboard, none of these heater entries appears, but their RPMs could be seen (eg blank entry against 2000 RPM which is for entity "Heater" in Remote. 
 
![2022-05-18-4](https://user-images.githubusercontent.com/62803887/169186406-6e2b2180-a9d8-4513-b333-7d7d8f4b83e8.JPG)

After our previous commit on Oct 7, I still could see blank against the Heater RPM under pump menu on dashpanel, but could reassign the heater to "Pool" or "Spa", as njsPC will successfully save the assignment.

New time-stamp and replay attached. 

![2022-05-18-3](https://user-images.githubusercontent.com/62803887/169185538-b795e331-ec6f-40ff-aa38-07fa2797d068.JPG)

[replay(4).zip](https://github.com/tagyoureit/nodejs-poolController/files/8722840/replay.4.zip)

**Comment by rstrouse** [comment, marked as answer]
Pull njsPC and dashPanel.  I have all the body references correct.  However, it appears the information we have for IntelliTouch virtual circuits is not correct and never has been.  Virtual Circuits are circuits that are internal to the OCP and include items like Heater, Pool Heater, Freeze.  There are a limited number of these on IntelliTouch but if you are game it would be great to get these all figured out correctly.

Based upon your excellent post above I was able to figure out the following.

```
155 = Spa/Hi-Temp Heater 
156 = Pool/Lo-Temp Heater
157 = Either Heater
```

However I would like to figure out the solar virtual circuit in IntelliTouch.  To do this could I get you to assign Valve A to solar from your Mobi or Indoor Control panel while in the middle of a packet capture.

I would be able to figure out the rest if you assign the following to a button on IS4 configuration.  You don't actually need the remote to do this and it would be greatly appreciated.  The only thing I need to know is which button you assigned to each.
```
Heat Boost
Pump Speed +
Pump Speed -
```

**Reply by vj950** [reply]
Thanks for the code update and working on fixing my issues. Seems to be working fine. And thanks for fixing issue #451, I was going to raise it once everything was working.

I am assuming virtual means virtual, so I actually have a Solar heater. I do not know how it's connected (Pool came with the house I purchased last year). I was told that It's been leaking somewhere and solar is now "disconnected". Physically, there is a Jandy actuator, the wiring goes in my OCP. The actuator itself is sitting on the ground and not attached to any valves. That's what I assume they mean by "disconnected". I will find time this weekend to see how it's hooked electrically. It seems that my Valve A is assigned to solar when I go in advanced settings on my remote. 

There is a dedicated Solar button on OCP. But when I start the "Solar" from dashpanel or from Remote, the light does not turn on. Therefore, I will have to see how the electrical connections are done (or are undone due to leak). Or may be it's meant to be virtual.
![20210923_214220](https://user-images.githubusercontent.com/62803887/169431054-139b99c7-e71d-480b-af5a-944a2cdd61d8.jpg)

Let me know how to proceed once I figure out valve connections. Also, I do not have IS4. Actually, I do not have any spa. That's why it's been very confusing for me understand how the programming is done. My "Spa" is actually my pool and "Pool" is solar. If there is another way to activate Pump speed +/- please let me know. Don't think there is any Heat Boost either.

**Reply by rstrouse** [reply]
That solar valve should be plugged into Valve A on the personality board.  Then you go into the valves configuration and set Valve A to Solar.   

There is a little 3 way switch that when it is in the middle the valve will not turn regardless of its selection on the OCP.  This keeps from filling the panels after they have been drained for the winter.  You can set this switch to the middle position so that it does not fill up the panels until you get it fixed.  

If they are Fafco panels they can leak around the rubber band clamps on the roof.  If Solaron, then there is a joint coupling that can leak but many times it is just the o-ring in the clamp.  The other place that can leak is the vacuum relief valve that is at the highest point on the solar.  This makes sure that if the panels drain down they fill with air instead of collapsing the pipes and tubing.  If they are Fafco or Solaron (most common) you can simply plug any of the tubes if they are leaking.  Other brands are less forgiving because they are not individual tubes.  Depending on your location joined tubing is not legal anymore because the wind can get under it and fly away.

The telltale sign of which panels you have is to notice whether they are connected together with band clamps.  If this is the case you likely have Fafco panels.  If there appears to be a special clamshell connector between the panels then these are likely Solaron.

**Comment by vj950** [comment]
attaching the replay for the following:
6.57pm Start "Solar (aka Master Pool)" circuit from Remote. Heater here is Solar Heater on my roof, therefore, the Jandy valve acutator turned by Intellitouch OCP to divert water to it.
6.58pm Turned "Solar" circuit off "Pool" (aka MasterSpa started), Solar valve actuator turned back to divert water to pool.

My Valve was already connected in OCP and was programmed to be used to turn water to Solar Heater on the roof. So I was unable to delete it using Remote, or add it. Before today, on all previous occasions of Replays, Jandy actuator was manually turned off by the switch on the actuator
[replay(6).zip](https://github.com/tagyoureit/nodejs-poolController/files/8775087/replay.6.zip)

I hope this is helpful. Otherwise I will have to read on installer menu to remove the valve assignment and add it again.

**Comment by rstrouse** [comment]
Is your solar valve attached to the plug labled VLVA on the i9+3S personality board?  If it is simply navigate to the valve configuration tab and set the solar circuit to the valve.  It should then use the solar sensor and water sensor to activate the valve.
![image](https://user-images.githubusercontent.com/47839015/170385333-6cac6218-5aea-4091-982d-366b9e5c94ed.png)

**Reply by rstrouse** [reply]
Also, unfortunately, the body circuit is also named solar so choose the second one.  It should only engage when it calls for heat.  I don't understand why the Lo-Temp body would have been renamed solar since this just confuses the whole situation.  Heck I am not even sure IntelliTouch would properly activate the valve when there is a Solar Heater enabled and the valve is assigned to anything other than the Solar virtual circuit.

**Reply by vj950** [reply]
> Also, unfortunately, the body circuit is also named solar so choose the second one. It should only engage when it calls for heat. I don't understand why the Lo-Temp body would have been renamed solar since this just confuses the whole situation. Heck I am not even sure IntelliTouch would properly activate the valve when there is a Solar Heater enabled and the valve is assigned to anything other than the Solar virtual circuit.

> Is your solar valve attached to the plug labled VLVA on the i9+3S personality board? If it is simply navigate to the valve configuration tab and set the solar circuit to the valve. It should then use the solar sensor and water sensor to activate the valve. 

That's how it was connected. And it works as intended now after I followed your instructions! I still have to figure out the leak situation, so the actuator is not actually connected to the valve yet. But it should work when it gets fixed! 

I will try to change the name Solar to something else. I really do not have another body to control from the same pump, so it's even more confusing. But it seems that after proper assignment per your instruction, it works fine for the pool. I will check with my Pool Service company, which happens to be the original installer of the pool (if they still have the same technicians on payroll).
